godtools/Managers/DownloadResources/DownloadedResourceManager.swift

```diff
@@ -52,79 +52,113 @@ class DownloadedResourceManager: GTDataManager {
+                    _ = BannerManager.shared.downloadFor(cachedResource!)
+                }
                 
-                cachedPage.filename = remotePage.filename
-                cachedPage.resource = cachedResource
+                let remoteTranslations = remoteResource.latestTranslations!
+                for remoteTranslationGeneric in remoteTranslations {
+                    let remoteTranslation = remoteTranslationGeneric as! TranslationResource
+                    let languageId = remoteTranslation.language?.id ?? "-1"
+                    let resourceId = remoteResource.id ?? "-1"
+                    let version = remoteTranslation.version!.int16Value
+                    
+                    if !translationShouldBeSaved(languageId: languageId, resourceId: resourceId, version: version) {
+                        continue;
+                    }
+                    
+                    var cachedTranslation = findEntity(Translation.self,byAttribute: "remoteId",withValue: remoteTranslation.id!)
+                    
+                    if cachedTranslation == nil {
+                        cachedTranslation = Translation()
+                        cachedTranslation!.remoteId = remoteTranslation.id!
+                        realm.add(cachedTranslation!)
+                    }
+                    
+                    cachedTranslation!.version = remoteTranslation.version!.int16Value
+                    cachedTranslation!.isPublished = remoteTranslation.isPublished!.boolValue
+                    cachedTranslation!.manifestFilename = remoteTranslation.manifestName
+                    cachedTranslation!.localizedName = remoteTranslation.translatedName
+                    cachedTranslation!.localizedDescription = remoteTranslation.translatedDescription
+                    
+                    cachedTranslation!.downloadedResource = cachedResource
+                    cachedResource!.translations.append(cachedTranslation!)
+                    
+                    let cachedLanguage = findEntity(Language.self, byAttribute: "remoteId", withValue: languageId)
+                    cachedLanguage?.translations.append(cachedTranslation!)
+                    cachedTranslation!.language = cachedLanguage
+                    
+                    TranslationsManager.shared.purgeTranslationsOlderThan(cachedTranslation!)
+                }
             }
         }
-        
-        saveToDisk()
     }
 
     private func translationShouldBeSaved(languageId: String, resourceId: String, version: Int16) -> Bool {        
         let predicate = NSPredicate(format: "language.remoteId = %@ AND downloadedResource.remoteId = %@",
                                     languageId,
                                     resourceId)
         
-        let existingTranslations: [Translation] = findEntities(Translation.self, matching: predicate)
+        let existingTranslations = findEntities(Translation.self, matching: predicate)
```

Comment:
I've thought about that, but one downside could be potential retain loops if we aren't careful. Since all of the manager classes share the same underlying find* methods, I think this is okay. Outside the manager classes, then we should use the appropriate manager class.